Reaqtor/Core/Reactive/Tests.Reaqtor.Reactive.Linq.Hosted/Distinct.cs

```diff
@@ -0,0 +1,326 @@
+                    OnNext(215, 3), //*
+                    OnNext(220, 3),
+                    OnNext(225, 2),
+                    OnNext(230, 2),
+                    OnNext(235, 1), //*
+                    OnNext(240, 2),
+                    OnCompleted<int>(250)
+                );
+
+                var res = client.Start(ctx, () =>
+                    xs.Distinct(),
+                    100, 200, 1000);
+
+                res.Messages.AssertEqual(
+                    OnNext(210, 2),
+                    OnNext(215, 3),
+                    OnNext(235, 1),
+                    OnCompleted<int>(250)
+                );
+
+                xs.Subscriptions.AssertEqual(
+                    Subscribe(200, 250)
+                );
+
+                Assert.AreEqual(0, env.ArtifactCount);
+            });
+        }
+
+        [TestMethod]
+        public void Distinct_Environment_KeySelector_Div2()
+        {
+            Run(client =>
+            {
+                var env = new TestExecutionEnvironment();
+
+                var ctx = client.CreateContext(env);
+
+                var xs = client.CreateHotObservable(
+                    OnNext(150, 1),
+                    OnNext(210, 2), //*
+                    OnNext(220, 4),
+                    OnNext(230, 3), //*
+                    OnNext(240, 5),
+                    OnCompleted<int>(250)
+                );
+
+                var res = client.Start(ctx, () =>
+                    xs.Distinct(x => x % 2),
+                    100, 200, 1000);
+
+                res.Messages.AssertEqual(
+                    OnNext(210, 2),
+                    OnNext(230, 3),
+                    OnCompleted<int>(250)
+                );
+
+                xs.Subscriptions.AssertEqual(
+                    Subscribe(200, 250)
+                );
+
+                Assert.AreEqual(0, env.ArtifactCount);
+            });
+        }
+
+        [TestMethod]
+        public void Distinct_Settings_MaxDistinctElements()
+        {
+            Run(client =>
+            {
+                var env = new TestExecutionEnvironment();
+
+                var ctx = client.CreateContext(env, settings: new Dictionary<string, object>
+                {
+                    { "rx://operators/distinct/settings/maxDistinctElements", 3 },
+                });
+
+                var xs = client.CreateHotObservable(
+                    OnNext(150, 1),
+                    OnNext(210, 2),
+                    OnNext(220, 3),
+                    OnNext(230, 4),
+                    OnNext(240, 5),
+                    OnCompleted<int>(250)
+                );
+
+                var res = client.Start(ctx, () =>
+                    xs.Distinct(),
+                    100, 200, 1000);
+
+                res.Messages.AssertEqual(
+                    OnNext(210, 2),
+                    OnNext(220, 3),
+                    OnNext(230, 4),
+                    OnCompleted<int>(250)
```

Comment:
We'll expect an error here, similar to https://github.com/reaqtive/reaqtor/blob/main/Reaqtive/Core/Tests.Reaqtive.Linq/Operators/Buffer.cs#L98.